src/devices/Common/Iot/Device/Common/ValueArray.cs

```diff
@@ -91,22 +91,53 @@ public int Count
         /// <inheritdoc />
         public bool IsReadOnly => false;
 
+#if NET5_0_OR_GREATER
         /// <summary>
         /// Returns a span pointing to this instance
         /// </summary>
         /// <returns>A span representing this instance</returns>
+        /// <remarks>
+        /// Due to a shortcoming of the Garbage collector, it is the caller's responsibility
```

Comment:
Nit: not really a shortcoming of the GC, rather the language (no escape semantics yet).